### PR TITLE
s3_wait: handle netsplits & add reusable poller

### PR DESCRIPTION
**Pull request by danielnorberg**
Continuation of #347

### CODE REVIEW

**Comment by danielnorberg** [comment]
@komamitsu PTAL

File: digdag-standards/src/main/java/io/digdag/standards/operator/PollingWaiter.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+        Config taskState = pollState.getNestedOrSetEmpty(TASK);
+
+        Optional<T> result;
+
+        try {
+            result = f.apply(taskState);
+        }
+        catch (Exception e) {
+            throw Throwables.propagate(e);
```
Comment (komamitsu) [inline digdag-standards/src/main/java/io/digdag/standards/operator/PollingWaiter.java]: We don't need to retry when an exception occurs?

File: digdag-standards/src/main/java/io/digdag/standards/operator/PollingWaiter.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        }
+
+        if (!result.isPresent()) {
```
Comment (komamitsu) [inline digdag-standards/src/main/java/io/digdag/standards/operator/PollingWaiter.java]: It seems that this class can be merged with PollingRetryExecutor that has many codes by adding a new predicate that tests a result to replace this line.

File: digdag-standards/src/main/java/io/digdag/standards/operator/PollingWaiter.java (hunk 1 above)
Comment (danielnorberg) [inline digdag-standards/src/main/java/io/digdag/standards/operator/PollingWaiter.java]: My intent was to have the `PollingRetryExecutor` handle retries separately from the `PollingWaiter` and keep waiting and retrying logic and state separate.

**Comment by komamitsu** [comment]
👍

**Pull request closed without merging**